src/evo/creditpool.cpp

```diff
@@ -321,3 +293,30 @@ bool CCreditPoolDiff::ProcessLockUnlockTransaction(const CTransaction& tx, TxVal
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "failed-procassetlocksinblock");
     }
 }
+
+std::optional<CCreditPoolDiff> GetCreditPoolDiffForBlock(const CBlock& block, const CBlockIndex* pindexPrev, const Consensus::Params& consensusParams,
+                                                         const CAmount blockSubsidy, BlockValidationState& state)
+{
+    AssertLockHeld(cs_main);
```

Comment:
why does this need cs_main?

Reply:
removed. Seems just inherited from similar code around, I don't see a reason why it is needed.